util/chplenv/chpl_bin_subdir.py

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env python
+
+import optparse
+import os
+from sys import stderr, stdout
+import sys
+
+chplenv_dir = os.path.dirname(__file__)
+sys.path.insert(0, os.path.abspath(chplenv_dir))
+
+import chpl_home_utils
+import chpl_platform, chpl_compiler, chpl_machine, chpl_arch
+
+from utils import memoize
+
+@memoize
+def get(location):
+    platform = chpl_platform.get(location)
+    compiler = chpl_compiler.get(location)
+    machine = chpl_machine.get(location)
+    (flag, arch) = chpl_arch.get(location, map_to_compiler=True,
+                                 get_lcd=chpl_home_utils.using_chapel_module())
+
+    # platform
+    result = platform
+    # compiler -- included for target
+    if location == 'target':
+      result += '-' + compiler
```

Comment:
I still have a slight preference for including compiler for host portions

Reply:
We talked about having it be `bin/linux64-x86_64/chpl`.. Adding the compiler would make it `bin/linux64-gnu-x86_64/chpl` and I thought we didn't want that. If this is really the path for a *bin* directory it shouldn't matter what compiler was used to make it, since it's not libraries (static or dynamic) but rather standalone programs.

Reply:
The only reason I put the *bin* directory here was because it matched what we were doing for third-party runtime dirs (so I could move most of the logic in default_uniq_cfg_path to here). But we could handle that separately if we were willing to change the order. Or just not call this from default_uniq_cfg_path.

Reply:
Michael convinced me offline to leave compiler out of host portions.

Reply:
My main concern was that people will find the -gnu there confusing and interpret it to mean that the compiler is configured to use GCC as the C backend. Additionally I mentioned that there are other ways to get toggling CHPL_HOST_COMPILER to trigger relinking. (And that appears to work now without any change).